src/addons/mod/assign/components/submission/addon-mod-assign-submission.html

```diff
@@ -162,25 +162,34 @@ <h2>{{ 'addon.mod_assign.userswhoneedtosubmit' | translate: {$a: ''} }}</h2>
                         </ng-container>
                     </ng-container>
 
-                    <!-- Add or edit submission. -->
+                    <!-- Edit and remove buttons. -->
+                    <div *ngIf="loaded && !isSubmittedForGrading && canEdit && !unsupportedEditPlugins.length &&
+                            !showErrorStatementEdit && (editedOffline || !removedOffline && userSubmission?.status &&
+                                userSubmission?.status !== statusNew && userSubmission?.status !== statusReopened)"
```

Comment:
The code works fine, this is just a suggestion. I prefer adding parenthesis when mixing different operators, even if they aren't needed, to make more clear the execution order.

```suggestion
                            !showErrorStatementEdit && (editedOffline || (!removedOffline && userSubmission?.status &&
                                userSubmission?.status !== statusNew && userSubmission?.status !== statusReopened))"
```

Reply:
Changed